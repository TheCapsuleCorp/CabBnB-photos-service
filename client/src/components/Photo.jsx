import React from 'react';

const Photo = (props) => {
  let photoUrl = props.photoUrl ? props.photoUrl.url : null;

  var divStyle = {
    backgroundImage: 'url(' + photoUrl + ')',
  };

  return (
    <div className={props.photoClass} style={divStyle}>
      {props.children}
    </div>
  );
};

export default Photo;
