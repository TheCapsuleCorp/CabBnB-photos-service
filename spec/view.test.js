import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Photo from '../client/src/components/Photo.jsx';

const photoUrl = 'https://cap-bnb-photo-service.s3.us-east-2.amazonaws.com/1-stream-belmont-3.jpg';

configure({ adapter: new Adapter() });

// describe('Photos', () => {
//   test('should render Photo component', () => {
//     const output = shallow(
//       <Photo photoClass={'leftPhoto'} photoUrl={photoUrl} />
//     );
//     expect(shallowToJson(output)).toMatchSnapshot();
//   });
// });
