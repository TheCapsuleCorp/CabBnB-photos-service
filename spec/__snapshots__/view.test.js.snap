// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos should render Photo component 1`] = `
<div
  className="leftPhoto"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
/>
`;
